graphene_federation/inaccessible.py

```diff
@@ -0,0 +1,54 @@
+from typing import Callable, Any, Optional
+
+from graphene import Schema, Field
+from graphene.types.schema import TypeMap
+
+
+def get_inaccessible_types(schema: Schema) -> dict[str, Any]:
+    """
+    Find all the inaccessible types from the schema.
+    They can be easily distinguished from the other type as
+    the `@inaccessible` decorator adds a `_inaccessible` attribute to them.
+    """
+    inaccessible_types = {}
+    for type_name, type_ in schema.graphql_schema.type_map.items():
+        if not hasattr(type_, "graphene_type"):
+            continue
+        if getattr(type_.graphene_type, "_inaccessible", False):
+            inaccessible_types[type_name] = type_.graphene_type
+    return inaccessible_types
+
+
+def inaccessible(field: Optional[Any] = None) -> Any:
+    """
+    Decorator to use to inaccessible a given type.
+    """
+
+    # noinspection PyProtectedMember,PyPep8Naming
+    def decorator(Type):
+        # TODO Check the provided fields actually exist on the Type.
+        # Set a `_inaccessible` attribute to be able to distinguish it from the other entities
+        setattr(Type, "_inaccessible", True)
+        return Type
+
+    if field:
+        field._inaccessible = True
+        return field
```

Comment:
I think this should be `if field: return decorator(field)`, no?

Reply:
Both the decorator and this check are adding an '_inaccessible' attribute. 

This marked usage (`if field` check) is satisfied in cases where individual fields are marked inaccessible.
eg: `out_stock = inaccessible(graphene.Boolean(required=True))`

Currently, the decorator is used for decorating Graphene Types and is defined as `decorator(type_)`. Its argument is named `type_` and one may expect the arg to be a graphene type and not a field.
Usage eg: 
```
@inaccessible
class PricingDetails(graphene.ObjectType):
    total = graphene.Int(required=True)
```

Shall we use the decorator in both cases or stick with the current implementation?

Reply:
I think field and type can be renamed to field_or_type if this decorator works for both cases :)

Reply:
Done 👍